### DISCUSSION TITLE
Network connection

### DISCUSSION THREAD

**Discussion by bengtner** [wlcrs/huawei_solar#102] [Setting up the integration]
I have my inverter hooked up on my network by means of the Huawei SDongleA WiFi dongle. I manage to connect to the maintenance port by using the FusionSolar app. Once connected I can open the Inverter WLAN settings, but I can't find any option to enable ModBus over TCP. Maybe this option has been removed by Huawei?

What puzzles me a bit is there is only config option, and this is for the Inverter WLAN settings that refers to a separate WLAN network. I guess this is the built in WLAN network is a different network that also gives access to the management port. But I suppose the app connects me to the management port via cloud access (using the Dongle). I have tried to connect to this network directly without success. Will require a WPA2 password. 

I guess this ModBus option is not available any longer. Anyone that can confirm this is the case?

**Comment by franblade** [comment]
Which is your inverter? Mine is SUN2000-5KTL-L1 and it's connected to my router and the integration is perfectly working.

**Reply by bengtner** [reply]
I have the same inverter, but this option is most likely controlled by the firmware. So a firmware update could disable this function. To be honest there are good reasons for Huawei to disable this option, since it is a nightmare from security point of view.

**Comment by wlcrs** [comment]
Can you share the exact firmware versions of your inverter and SDongle. Can you make screenshots of what you see in the FusionSolar app?

**Comment by bengtner** [comment]
I have a SUN2000-10KTL inverter. I manage to connect to the management interface either through the dongle (the inverter is hooked up on my LAN with a SDongleA-05) or through the built in WiFi access point using the FusionSolar app. There are just one option avaliable - Communucation configuration. Under that tab I can just find Inverter WLAN settings where I can change the password for the build in WLAN. Nothing more. It doesn't matter if I connect by the dongle or the WLAN. Just this option available. And absolutely nothing about Modbus ober TCP.

Looking in the Maintenance tab under Upgrade device, I find option Donge upgrade, Current version V100R001C00SPC119. I get a feeling this refers to the inverter firmware even if the tab refers to "Dongle upgrade". I can't find any other version info in the app. There is an option to download logs beside this.

Edit: I must admit I'm extremely confused now and I'm not 100% sure I have logged in to the management system over the inverter WLAN. I thought I did, but now the default password (with username Installer) does not work. So I can connect to the SUN2000 WLAN, but I get "wrong password" if I use the same as I do with cloud access through the dongle. Are there actually 2 different managment systems? This could explain my problem. What I have described above is when I access the management system from cloud via the dongle. And in this case there are option to enable Modbus over TCP. If there is a separate management system behind the inverter WLAN, I might this option there. However, the password seems to a different one since the default (starting with 000...) does not work.

Edit2: Right, there are actually two different management systems! I made a reset of the password for the inverter management system (accessed over the built in WLAN). And now I have a completeley different view in the app and I've managed to set the Modbus over TCP option (unrestricted). For some reason I made a mistake and thought I was connected over the built-in WLAN, but probably I was still connected over cloud and the dongle. So now the fundamental setting should be OK. However, when I try to initiate the Huawei Solar integration and specify the Ip address given to the dongle on my LAN, still it fails. I try to use port 502 right? I have SPC119 in my dongle and SPC141 in the inverter.

**Comment by bengtner** [comment]
Thanks for the excellent wiki! Unfortunately I didn't see this until now....

I have one question about optionA. Is it possible to use the dongle in parallel with RS485? Someone told me the dongle will act as a Modbus master, and must be removed if you connect another Modbus master to the inverter. Eventually HA will give me excellent statistics, but probably it will take same time before I get there. Meanwhile it would be nice to keep FusionSolar to keep track of the daily yield.

**Reply by wlcrs** [reply]
I don't have a dongle myself, but this person does: https://github.com/wlcrs/huawei_solar/issues/105#issuecomment-1200524588

**Comment by bengtner** [comment]
Finally I have working setup using the dongle for network connectivity! It required a firmware update requested from Huawei support organisation. Now I have SPC 146 on the inverter and SPC130 on the dongle. I followed the guidelines for the configuration. However, I had to specify Slave Id: 1 to make it work. I think it would be good to update the wiki with info about needed firmware update even if this is a moving target. I just wonder why Huawei does not provide those versions for the guys doing the initial commisioning. In my case he made an update of the software, but obviously not to the required versions to make Modbus over TCP working over the dongle.

Now I can read various sensors and use those in my Home Assistant setup. However, I have one remaining problem with the log filled up with the following lines:
```
Logger: pymodbus.client.asynchronous.async_io
Source: /usr/local/lib/python3.10/site-packages/pymodbus/client/asynchronous/async_io/__init__.py:307
First occurred: August 17, 2022 at 21:52:44 (155 occurrences)
Last logged: 09:49:35

Failed to connect: [Errno 111] Connect call failed ('192.168.0.47', 6607)
Failed to connect: Multiple exceptions: [Errno 111] Connect call failed ('::1', 6607, 0, 0), [Errno 111] Connect call failed ('127.0.0.1', 6607)
```
It seems like the integration fails to read some data. I just wonder why and what data this is. Or is it just a data communication hickup?

I have a SmartMeter DTSU666. Is it possible to readout my local power consumption in my house with this integration?  I've read a few posts about this, but they are from 2021 or older. 

Edit: I've just discovered the Modbus cable is not (yet) connected to the Smartmeter. I guess this a still on the providers "to-do" list. I'm still waiting for the battery pack. So this will probably be solved sooner or later. This might be the root cause for log entries as well. I have to be a bit more patient

**Comment by alexualbu** [comment]
did you ever manage to fix this? i have similar issues, except in my case it doesn't succeed to read anything, the connection gets interrupted before (and then i can also see the errors you mentioned in the last post)

**Comment by TMoersch** [comment]
Hi alexualbu,
I do have have a different Huawei Inverter but my configuration works.
0. Update all Firmware components such as Dongle, Inverter and Battery if necessary
1. Step connect the DTSU with twistetpair connection to inverter. ( or let it do because if this connection missing the inverter cant react to smater meter data)
2. go to FusionSolar and enable MobusTCP ==>set it to the IP of raspy or to unrestricted make sure you press Upadte

This is only available with the installer account !! So you maybe have to connect your solar installer?
Step 0 is the root to make it possible
Step1 makes the smartmeter data visible in the inverter
Step2 allows external, but local systems like HomeAssitent/evcc to connect to the inverter via modbus tcp...

Thomas

**Reply by alexualbu** [reply]
Thanks, @TMoersch . It worked  out in the end - there seems to be a bug in the integration (or pymodbus) that causes it to keep wanting to reconnect after it fails. I tried with slave_id 0 as suggested and it's actually 1, but subsequent retries would not work. After a restart of HA and the right slave_id it worked.
